### DISCUSSION TITLE
I'm having an issue getting my application approved

### DISCUSSION THREAD

**Discussion by thedewpoint** [etsy/open-api#675] [Q&A] [answered]
I've been rejected several times, most recently for "Generated API key at the request of 3rd party app" which I have no idea what it means. This is the only step holding me up from finishing my app that I'm building. Can anyone help me understand why I'm being rejected?

**Comment by thedewpoint** [comment]
Also, annoyingly I cannot see the reason I was denied until I submit another application which ends up being a waste because I'm making the same mistakes again.

**Reply by hessodreamy** [reply]
I know! It's so stupid! Sorry, not helpful, I'm having a similar issue.

**Reply by etsyachristensen** [reply]
@thedewpoint Just to get on the same page, you could not see the "Generated API key at the request of 3rd party app" message until after you submitted a second application? It suddenly showed up then?

**Reply by thedewpoint** [reply]
@etsyachristensen yea unfortunately when you get rejected there is no way to see why you were rejected unless you submit another application. I'm kind of spamming the system because of it, although the last 5 or 6 times were all for the same reason. I am kind of shooting in the dark now trying to figure out the real reason why.

**Reply by etsyachristensen** [reply]
Ok thank you. That should not be the case. We will get into it and fix that.

**Comment by etsyachristensen** [comment, marked as answer]
@thedewpoint . The message you are getting indicates that the person reviewing your application believes that you are creating an app to get an API Key to enter into a 3rd party software system. This is strictly forbidden by our terms of service. If you are indeed creating your own app and not providing your API credentials to another party, then when you re-apply please describe in detail what your app does and how it will be used.

**Reply by thedewpoint** [reply]
Hey @etsyachristensen thanks so much for replying. I'm not requesting anyone generate anything I'm planning to use the Oauth flow but I can't release it because I can't finish building it (because I can't get approved) Right now I have the OAuth flow but when I try to grant access to my application it's not a recognized application (because I can't get approved) so I can't get a token. It's kind of a chicken and egg problem. Any advice on how to proceed?

**Reply by etsyachristensen** [reply]
Sorry @thedewpoint that's not what I meant. It's the other way around. The app description, urls or something was indicating that some other system was asking you to generate an app in Etsy to get an api key to put into their system to work on your behalf. Likely as not, the issue comes down to the description or something. Could you provide the name of the application you applied for that was rejected so we can look it up and try to get more info on why the reviewer felt it was against policy?

**Reply by thedewpoint** [reply]
Thank you!
Sellertools - Catalog Sync

**Reply by etsyachristensen** [reply]
Looks like the most recent application has been approved. Sorry for all the trouble.

**Reply by thedewpoint** [reply]
thank you!!

**Comment by Ypatel2607** [comment]
Hi @etsyachristensen ,

I'm running into the same issue and hoping you can help.

I built an app called "compliscan" — it's an e-commerce compliance scanner that helps Etsy sellers check their listings for regulatory issues like California Prop 65, EU GPSR, CPSIA, and FDA labeling requirements. The app is live at [.](https://compliscan.app/)

I've submitted two applications, and both were banned. The first rejection mentioned "does not clearly explain the app's use case" and "appears to be submitted on behalf of a third-party app." The second rejection came with no explanation at all.

To be clear about what the app does:

- It's my own app, not generating keys for another service
- It uses OAuth v3 with PKCE (Etsy's recommended auth flow) — no manual API key sharing
- It only requests listings_r scope (read-only access to listings)
- Sellers connect their shop, we scan their listing titles and descriptions against regulatory databases, and show them which products have compliance gaps with exact fix language
- The seller benefits directly — avoiding fines (Prop 65 fines start at $2,500/violation/day) and marketplace takedowns

I'm not sure what in the description triggered the "third-party app" flag. Could you look up the application and help me understand what needs to change? I'm happy to provide any additional detail.

Thank you,
Yash Patel
